DOC updated the docs on the bug release (#486)
paris-saclay-cds/ramp-board: doc/contribute.rst

    and `.. include::` this new file in `doc/whats_new.rst
    <https://github.com/paris-saclay-cds/ramp-board/blob/master/doc/whats_new.rst>`_.

* Note that the steps 4, 5 and 7 should be performed while in the release
branch, e.g. branch `0.5.X`.

Patch/bug fix release process
-----------------------------

1. Checkout the tag for the last minor release, e.g.,
   `git checkout 0.5.X`.
   **note**: X is part of the name, not a number to substitute.
   **note2**: you can type `git fetch --all --tags` to fetch available tags.
2. There are two ways to continue. If you only want to release a couple of

>    commits we recommend that you go with '3a', if you want to release many commits '3b' would be easier.
>    to continue with '3b'.

Reviewer: ```suggestion
   commits we recommend that you go with '3a', if you want to release many commits '3b'
   would be easier.
```

3a: release only a few commits

- Find the commit(s) hash of the bug fix commit you wish to back port using
  `git log`.
- Append the bug fix commit(s) to the branch using `git cherry-pick <hash>`.

3b: release multiple commits

- checkout the `0.5.X` branch:
  `git checkout -b release-0.5.<number> master` where 'number' is the number of
  your bug release.
- commit your desired changes by typing: `git rebase -i master`.
  You should now see the list of all the commits since the last release.
  Indicate which commits you want to release by typing 'pick' on their lefthand
  side, and 'drop' if you do not want to keep it. Make sure to drop the commit
  with previous bumpversion. It's useful to have a copy of the `git rebase -i`
  log in the PR to help others understand what's included.

> - make a PR to the master. Once it is merged continue to the next step.

Reviewer: > Do you PR to master or the release branch?

I also think it was meant to be release

```suggestion
- make a PR to the release branch. Once it is merged continue to the next step.
```

> Also if you need to make PR here do you also need to make PR at the end of 3a?

I guess it depends on the person doing it. If it's just a commit, cherry-picking directly to the release branch sounds also fine. A PR is mostly useful to discuss what to pick or exclude.

1. Checkout the tag for the last minor release, e.g.,
   `git checkout 0.5.X`.
   **note**: X is part of the name, not a number to substitute.
   **note2**: you can type `git fetch --all --tags` to fetch available tags.
2. There are two ways to continue. If you only want to release a couple of

>    commits we recommend that you go with '3a', otherwise the easier will be

Reviewer 2: ```suggestion
   commits we recommend that you go with '3a', if you want to release many commits '3b' would be easier.
```

   to continue with '3b'.

3a: release only a few commits

- Find the commit(s) hash of the bug fix commit you wish to back port using
  `git log`.
- Append the bug fix commit(s) to the branch using `git cherry-pick <hash>`.

3b: release multiple commits

> - being on the `0.5.X` checkout a branch:

Reviewer 2: ```suggestion
- checkout the `0.5.X` branch:
```

  `git checkout -b release-0.5.<number> master` where 'number' is the number of
  your bug release.
- commit your desired changes by typing: `git rebase -i master`.
  You should now see the list of all the commits since the last release.
  Indicate which commits you want to release by typing 'pick' on their lefthand
  side, and 'drop' if you do not want to keep it. Make sure to drop the commit

>   with previous bumpversion. It's usefult to have a copy of the `git rebase -i`

Reviewer 2: ```suggestion
  with previous bumpversion. It's useful to have a copy of the `git rebase -i`
```

  log in the PR to help others understand what's included.

> - make a PR to the master. Once it is merged continue to the next step.

Reviewer 2: Do you PR to master or the release branch? Also if you need to make PR here do you also need to make PR at the end of 3a?
